language/move-prover/bytecode/src/borrow_analysis.rs

```diff
@@ -590,9 +590,14 @@ impl<'a> TransferFunctions for BorrowAnalysis<'a> {
                         let callee_target = &self
                             .targets
                             .get_target(callee_env, &FunctionVariant::Baseline);
-                        if let Some(callee_an) =
+                        let native_an;
+                        let callee_an_opt = if callee_env.is_native() {
+                            native_an = native_annotation(callee_env);
```

Comment:
Not sure why you need to do this, but before the native function was visited by the `process` function which initialized the borrow annotation, so it could be retrieved here without special casing. Is the "fake" native function not longer appearing in the list of module functions, and thus `process` never called? This would explain why you need to do this here.

Reply:
Right now the native function is still in the module functions. But maybe it isn't getting marked as a dependency somehow?
Is that process done somewhere else? 

I guess what I'm trying to say is I suspect that `Vector::borrow_mut` isn't being visited beforehand. And I do not know what spot of code would be causing that to happen.

Reply:
@wrwg, I'll take your approval as a sign that you are fine with this for now. Let me know if you want me to fix it. Also happy to do so after this lands, if that is easier :)

Reply:
While I still think it's better to keep all those guys well-known functions instead of turning them into bytecodes, I'm OK to merge this. I can come back to this and turn the functions into `Operation::VectorBorrow` etc. this will then resolve this issue here.